homeassistant/components/gpstracker/config_flow.py

```diff
@@ -0,0 +1,135 @@
+"""Config flow for GPS Tracker integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohttp
+import gps_tracker
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+from .helpers import get_gpstracker_client
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    cfg = gps_tracker.Config(  # type: ignore[call-arg]
+        password=data[CONF_PASSWORD],
+        username=data[CONF_USERNAME],
+    )
+
+    async with get_gpstracker_client(hass, cfg) as client:
+        try:
+            await client.get_devices()
+        except gps_tracker.client.exceptions.UnauthorizedQuery as err:
+            raise InvalidAuth from err
+        except (
+            gps_tracker.client.exceptions.HttpException,
+            aiohttp.ClientResponseError,
+        ) as err:
+            raise CannotConnect from err
+
+
+class GpsTrackerFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for GPS Tracker."""
+
+    VERSION = 1
+
+    entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await validate_input(self.hass, user_input)
+            await self.async_set_unique_id(user_input[CONF_USERNAME].lower())
```

Comment:
You should move these two lines out of try into else statement.

Reply:
Fixed in 6cd945738e66a249e8691c243d817e869028b682